api/pkg/s3/bucketlifecycleget_test.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2019 The OpenSDS Authors.
+
+//
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+
+// you may not use this file except in compliance with the License.
+
+// You may obtain a copy of the License at
+
+//
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+//
+
+// Unless required by applicable law or agreed to in writing, software
+
+// distributed under the License is distributed on an "AS IS" BASIS,
+
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+// See the License for the specific language governing permissions and
+
+// limitations under the License.
+
+package s3
+
+import (
+	"fmt"
+	"github.com/opensds/multi-cloud/s3/proto/mocks"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+	backend "github.com/opensds/multi-cloud/backend/proto"
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+
+	backendMock "github.com/opensds/multi-cloud/backend/proto/mocks"
+)
+
+func TestAPIService_tier2class(t *testing.T) {
+
+	s3ClientMock := &mocks.S3Service{}
+	s3ClientMock.On("GetStorageClasses", context.Background(), &s3.BaseRequest{}).Return(MockedGetStorageClasses()).
+		Once()
+
+	//mockS3Service := &S3ServiceMock{}
+	//mockS3Service.On("GetStorageClasses").Return(MockedGetStorageClasses())
+	backEndMock := &backendMock.BackendService{}
+
+	type fields struct {
+		s3Client      s3.S3Service
+		backendClient backend.BackendService
+	}
+	fields1:= fields{
+		s3ClientMock,
+		backEndMock,
+	}
+	type args struct {
+		tier int32
+	}
+	args2:=args{99}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{"test1",fields1,args2,"STANDARD_IA",false},
+		// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &APIService{
+				s3Client:      tt.fields.s3Client,
+				backendClient: tt.fields.backendClient,
+			}
+			got, err := s.tier2class(tt.args.tier)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("APIService.tier2class() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("APIService.tier2class() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+
+func BucketGetHandler(rq *restful.Request, rp *restful.Response) {
```

Comment:
Will it be good if we move these helper functions (non Test* functions) in a separate module so that it can be at common place and reusable?